File: Java/URI1021.java
import java.util.Scanner;

public class URI1021 {
	public static void main(String[] args) {
		Scanner leitor = new Scanner(System.in);
		float valor = leitor.nextFloat();
		float [] lista = {100,50,20,10,5,2,1, 0.50f,0.25f,0.10f,0.05f,0.01f};
		int temp = 0;
		valor += 0.001f;
		int [] result = new int [12];
		for (int k = 0; k < lista.length; k++) {
			temp = (int) (valor / lista[k]);
			result[k] = temp;
			valor -= temp * lista[k];
		}
		System.out.printf("NOTAS:\n%d nota(s) de R$ 100.00\n%d nota(s) de R$ 50.00\n%d nota(s) de R$ 20.00\n%d nota(s) de R$ 10.00\n%d nota(s) de R$ 5.00\n%d nota(s) de R$ 2.00\nMOEDAS:\n%d moeda(s) de R$ 1.00\n%d moeda(s) de R$ 0.50\n%d moeda(s) de R$ 0.25\n%d moeda(s) de R$ 0.10\n%d moeda(s) de R$ 0.05\n%d moeda(s) de R$ 0.01\n",result[0], result[1],result[2],result[3],result[4],result[5],result[6],result[7],result[8],result[9],result[10], result[11]);	
	}

}

File: C/uri1021.c
#include <stdio.h>
#include <math.h>

void main(){
    double valor, calc;
    scanf("%lf", &valor);
    double valores[12] = {100, 50, 20, 10, 5, 2, 1, 0.5, 0.25, 0.10, 0.05, 0.01};
    int k;
    for(k = 0; k < 12; k++){
        calc = valor/valores[k];
        if(k == 0) printf("NOTAS:\n");
        if(k == 6) printf("MOEDAS:\n");
        if(k < 6){
            printf("%d nota(s) de R$ %.2lf\n", (int) calc, valores[k]);
        }else if(k == 11){
            printf("%.0lf moeda(s) de R$ %.2lf\n", calc, valores[k]);
        }else{
            printf("%d moeda(s) de R$ %.2lf\n", (int) calc, valores[k]);
        }
        valor = fmod(valor,valores[k]);
    }
}